content/articles/compile-your-node.js-application-into-a-.exe-file/index.md

```diff
@@ -0,0 +1,537 @@
+
+> **NOTE:** As long as the executable file is running, the application is running.
+
+Close it using `Ctrl + C`.
+
+Now let us dive a little deeper and specify the operating system and the node version.
+In this, we use the `--targets` command.
+
+Delete the executable files generated in the previous process.
+
+Run the following in the Terminal:
+
+```bash
+pkg index.js --targets node12-win-x64
+```
+
+This specifies that the project should be compiled into an executable file that runs on Node version 12 and Windows O/S of a 64-bit architecture.
+
+Some of the supported node ranges, platforms, and architectures are as shown in the table below:
+
+| Node Versions | Platforms  | Archs  |
+|---|---|---|
+| node8  | alpine  | x64  |
+| node10  | linux  | arm64  |
+| node12  | linuxstatic  | (armv6)  |
+| node14  | win  | (armv7)  |
+| node16  | macos  |   |
+| latest  | (freebsd)  |   |
+
+#### Compile your project into an executable using pkg
+
+Here, we shall have to tell `pkg` our resources folder. We can do this inside the `package.json` file using scripts and assets configurations.
+Let us head over to our `package.json` file and add a "pkg" as shown below:
+
+```json
+{
+    "name": "executable",
+    "bin": "index.js",
+    "version": "1.0.0",
+    "description": "Simple express app",
+    "main": "index.js",
+    "scripts": {
+        "start": "node index.js",
+        "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "pkg": {
+        "assets": [
+            "views/**/*"
+        ],
+        "output": "dist"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "dependencies": {
+        "express": "^4.17.1",
+        "pkg": "^5.2.1"
+    }
+}
+```
+
+You can notice we have also added "bin". This tells `pkg` where the app entry point is. Inside the "pkg" script, we have added "assets" to show which file contains our resources. You can add more by just separating them with a comma.  We have further specified where the outputs will be stored using "outputPath".
+
+Run `pkg` using the command below to use our configuration in `package.json`:
+
+```bash
+pkg .
+```
+
+or
+
+```bash
+pkg package.json
+```
+
+In the case of build files or the specific targets, you can also specify them to pkg to obtain an output of the format below:
+
+```json
+{
+    "pkg": {
+              "scripts": "build/**/*.js",
+              "assets": "views/**/*",
+              "targets": [ "node14-linux-arm64" ],
+              "outputPath": "dist"
+            }
+}
+```
+
+#### File or directory not included Error
+
+Incase you might have encountered "was not included" Error in pkg, then this is how to get around it.
+In pkg module, we can not use direct concatenation of relative paths, for instance `res.sendFile(__dirname + '/views/index.html');`. This is because it is not only a **bad programming practice** but also because it will retun an error as shown below:
```

Comment:
```suggestion
In pkg module, we shall not use a direct concatenation of relative paths, for instance `res.sendFile(__dirname + '/views/index.html');`. This is because it is not only a **bad programming practice** but also because it will return an error as shown below:
```